### PR TITLE
Display error in Dashboard when user account is not present, remove warnings, minor formatting

### PR DESCRIPTION
**Pull request by cellog**
removes warnings in console on viewing provider page, and on viewing dashboard. A comment is made more readable as well.

The dashboard now displays an error on user account not being present, after a 500ms delay (which is configurable)

screen shot 2018-12-18 at 3 40 09 pm

Component is fully tested and has a new story under dashboard to show the error state.

### CODE REVIEW

**Comment by julien51** [comment]
I think the warning is actually legit and we should require an account to be passed to the Dashboard :(

The correct course of action here is probably to not render the Dashboard until an account is passed to it. What do you think?

**Comment by cellog** [comment]
That is basically what happens now, but I guess you're suggesting that the bail out be in the dashboard page instead of the component?

**Comment by julien51** [comment]
Yes exactly, we should not invoke `<Dashboard />` without passing it an actual account.

**Comment by cellog** [comment]
@julien51 ready for final review! (I hope)

File: unlock-app/src/components/helpers/SuspendedError.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    delay: 200,
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      show: false,
+    }
+    this.mounted = false
+    this.clearTimeout = false
```
Comment (julien51) [inline unlock-app/src/components/helpers/SuspendedError.js]: This will make typechecking upset later :/

File: unlock-app/src/pages/dashboard.js [hunk 2]
```diff
@@ -30,13 +41,14 @@ export const Dashboard = ({ account, network, locks }) => {
...
 Dashboard.propTypes = {
-  account: UnlockPropTypes.account.isRequired,
+  account: UnlockPropTypes.account,
   network: UnlockPropTypes.network.isRequired,
   locks: UnlockPropTypes.locks,
 }
 
 Dashboard.defaultProps = {
   locks: {},
+  account: null,
```
Comment (julien51) [inline unlock-app/src/pages/dashboard.js]: no need, right?

File: unlock-app/src/pages/dashboard.js [hunk 3]
```diff
@@ -4,14 +4,25 @@ import NoSSR from 'react-no-ssr'
...
 
 export const Dashboard = ({ account, network, locks }) => {
   if (!account) {
-    return null //loading
+    return (
+      <SuspendedError>
+        <DefaultError title="User account not initialized">
+          <p>
+            In order to display your Unlock dashboard, you need to connect a
+            crypto-wallet to your browser
```
Comment (julien51) [inline unlock-app/src/pages/dashboard.js]:
```suggestion
            crypto-wallet to your browser.
```

**Comment by cellog** [comment]
@julien51 one more review and I think we are done. 😩

**Pull request merged**